### ISSUE TITLE
[data grid] Aggregation Column Header Label Should be Optional

### ISSUE THREAD

**Issue by mschaefer-gresham** [mui/mui-x#5682]
### Current behavior 😯

Aggregation label in column header should be optional. Providing empty label doesn’t work because column titles become unaligned. 

Leaving the label field off the aggregation function results in the aggregation function name being used as the label.

Using an empty string for the label doesn't work because then the column title drops slightly and doesn't line up with non-aggregated columns.

### Expected behavior 🤔

It should be possible to not display the aggregation label in the column header and for column titles to line up.

### Steps to reproduce 🕹

Link to live example:

Steps:
1. Create custom aggregator function. 
2. Set label to empty string.
3. Add the aggregator to a column.
4. See that the column title of this column doens't line to other columns.

### Context 🔦

We don't want the aggregator label to be displayed in the column header.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
 System:
    OS: macOS 12.5
  Binaries:
    Node: 16.3.0 - ~/.nvm/versions/node/v16.3.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.20.0 - ~/.nvm/versions/node/v16.3.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.134
    Edge: Not Found
    Firefox: 86.0.1
    Safari: 15.6
  npmPackages:
    @emotion/react: ^11.10.0 => 11.10.0 
    @emotion/styled: ^11.10.0 => 11.10.0 
    @mui/base:  5.0.0-alpha.92 
    @mui/icons-material: ^5.8.4 => 5.8.4 
    @mui/lab: ^5.0.0-alpha.93 => 5.0.0-alpha.93 
    @mui/material: ^5.9.3 => 5.9.3 
    @mui/private-theming:  5.9.3 
    @mui/styled-engine:  5.8.7 
    @mui/system: ^5.9.3 => 5.9.3 
    @mui/types:  7.1.5 
    @mui/utils:  5.9.3 
    @mui/x-data-grid:  5.15.0 
    @mui/x-data-grid-generator: ^5.15.0 => 5.15.0 
    @mui/x-data-grid-premium: ^5.15.0 => 5.15.0 
    @mui/x-data-grid-pro:  5.15.0 
    @mui/x-license-pro:  5.15.0 
    @types/react: ^18.0.15 => 18.0.15 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    typescript: ^4.7.4 => 4.7.4 

```
</details>

### Order ID 💳 (optional)

45466

**Comment by cherniavskii** [comment]
Great, first feedback on the aggregation feature!
cc @flaviendelangle 

BTW @joserodolfofreitas this is a good case for "enhancement" label, since "new feature" label doesn't feel right here.

**Comment by flaviendelangle** [comment]
Hi,
Thanks for your contribution.

Could you provide a working reproduction, this [Codesandbox template](https://codesandbox.io/s/datagrid-v5-quick-start-54iz1) might be a good starting point.

____

What is your use case ?
Are you trying to hide all the aggregation label in the headers ?

**Comment by mschaefer-gresham** [comment]
[mui-testing.zip](https://github.com/mui/mui-x/files/9282887/mui-testing.zip)

**Comment by joserodolfofreitas** [comment]
I see an opportunity for this option when the grid is statically aggregating the data. 
In this case, it's probably clear to the users that the summary footer is always going to be a single aggregated function, for instance, `sum`.

Is that your case @mschaefer-gresham?

**Comment by mschaefer-gresham** [comment]
Can you please reopen this issue. I provided clear instructions for how to reproduce this issue and even a working reproduction (even though it couldn't be more straightforward to reproduce. Just remove the label from the custom aggregator example in the documenation). I also responded to the inqueries about our use case. 

We will always apply the same custom aggregator function for numeric columns and we disable the column settings menu in the header.  It must be possible to disable/hide the aggregator function label in the column header since (in our case) it is just not needed and is in fact blemish on the design of our application.

**Comment by cherniavskii** [comment]
Hey @mschaefer-gresham 
I've created a demo on Codesandbox from the one you provided just to make it easier to see it in action:
https://codesandbox.io/s/aggregationinitialstate-demo-mui-x-forked-r29xjk?file=/demo.tsx

**Comment by flaviendelangle** [comment]
OK
So the need here is to totally disable the aggregation label in the header for all columns ?
Or on a column by column basis ?

**Comment by mschaefer-gresham** [comment]
@flaviendelangle for our purposes we will always disable the aggregation label in all columns.

**Comment by flaviendelangle** [comment]
@cherniavskii we could add a new prop `disableAggregationHeaderLabel`
I don't see why someone would need to disable only on certain columns, the output would be weird.
Or at least that's enough of an edge case to not complexify the main api.

**Comment by cherniavskii** [comment]
@flaviendelangle What do you think about supporting `null` value for `label` to not render label element at all in this case?

**Comment by flaviendelangle** [comment]
That would be for a column by column customization right ?

**Comment by cherniavskii** [comment]
No, I meant `GridAggregationFunction['label']`, that would be for all columns (but for a single aggregation function)
But I just realized the same label is also used in column menu aggregation select :/

**Comment by cherniavskii** [comment]
We can hide the label with CSS though:
```tsx
<DataGridPremium
  sx={{
    ".MuiDataGrid-aggregationColumnHeaderLabel": {
      display: "none"
    }
  }}
/>
```

Working demo: https://codesandbox.io/s/aggregationinitialstate-demo-mui-x-forked-5mlj6l?file=/demo.tsx:2213-2408

**Comment by flaviendelangle** [comment]
> But I just realized the same label is also used in column menu aggregation select :/

Yeah, and that would also force users to override all the built-in aggregation methods to hide the label.

**Comment by overweightdev** [comment]
Hiding the label with CSS works for me, thanks! I think having an easy to use flag like disableAggregationHeaderLabel might be a nice convenience feature as well. 

Also just my opinion but it seems like a strange UI choice to have the aggregate function label in the column header. If a label is needed it might make more sense having it next to the aggregate value itself I wonder. Food for thought

**Comment by flaviendelangle** [comment]
The place where the label is rendered was the topic of endless discussion, and I think that everyone at MUI who worked on it saw that it was a pain to find the perfect place that match all use-cases.

We ended up with the header, with the idea of maybe allowing to place it somewhere else if the user prefer it on the cell itself or on the grouping column.

The cell itself has two main downsides:

1. With heavily-customized `renderCell`, it's complication to add a label
2. The column width would have to react to the presence of the label, otherwise, all the default width provided by the user would probably not be sufficient when a label is present

**Comment by DevPowers** [comment]
Whats the status on this? The aggregation label used to be hide-able but is recently showing back up again on my headers

**Comment by michelengelen** [comment]
Closing this in favor of #14388, which now tracks the design work for the aggregation label placement (`aggregationLabelPosition: "header" | "inline"`). The discussion in this thread anticipated exactly that direction — moving the label next to the aggregate value removes it from the header and fixes the title alignment. I noted your use case there, so a fully hidden variant is considered alongside the inline one.

For hiding the label today, the workaround from [this comment](https://github.com/mui/mui-x/issues/5682#issuecomment-1225857991) still works and keeps the column titles aligned:

```tsx
<DataGridPremium
  sx={{ '.MuiDataGrid-aggregationColumnHeaderLabel': { display: 'none' } }}
/>
```

A dedicated `disableAggregationHeaderLabel` prop never shipped, and with the reposition option on the table it would be redundant. Please add your 👍 and use cases to #14388.
